### PR TITLE
tree-wide: drop deprecated io/ioutil

### PR DESCRIPTION
**Pull request by mythi**
Go 1.16 release notes announced the deprecation of io/ioutil [1]. It's easy
for us to move to use what is was recommended so just do it.

[1] https://golang.org/doc/go1.16#ioutil

Closes: #612 

Signed-off-by: Mikko Ylinen <mikko.ylinen@intel.com>

### CODE REVIEW

**Review by bart0sh — DISMISSED** [review]
/lgtm

**Comment by eero-t** [comment]
This seems a bit pre-mature as this change is supported only from Go v1.16 onward.

(CI seems to be still at v1.15, Debian testing is still at v1.15, Ubuntu supports v1.16 only from 21.04 onwards (released last month), and e.g. Fedora 33 is also still at Go v1.15.)

**Comment by mythi** [comment]
> This seems a bit pre-mature as this change is supported only from Go v1.16 onward.

We have plans to move to v1.16 but need the base image conversion done first (so that we can pull `golang:1.16-buster`). I'll finish this and bump up the CI versions afterwards.

File: build/docker/intel-fpga-initcontainer.Dockerfile [hunk 1]
```diff
@@ -1,73 +1,64 @@
...
-ln -sf $HOOK_CONF_DST /etc/containers/oci/hooks.d/$HOOK_CONF\n\
-rm $DST_DIR/deploy.sh\n\
-">> $SRC_DIR/deploy.sh && chmod +x $SRC_DIR/deploy.sh
+ARG TOYBOX_VERSION="0.8.4"
+RUN apt update && apt -y install musl musl-tools musl-dev
+RUN curl -SL https://github.com/landley/toybox/archive/refs/tags/$TOYBOX_VERSION.tar.gz | tar xz \
+    && cd toybox-$TOYBOX_VERSION \
+    && KCONFIG_CONFIG=${DIR}/build/docker/toybox-config LDFLAGS="--static" CC=musl-gcc PREFIX=$ROOT V=2 make toybox install \
+    && install -D LICENSE $ROOT/usr/local/share/package-licenses/toybox \
+    && cp -r /usr/share/doc/musl $ROOT/usr/local/share/package-licenses/
```
Comment (eero-t) [inline build/docker/intel-fpga-initcontainer.Dockerfile:59]: This and other initcontainer dockerfiles changes are pulling source tarball from internet without checking it matches what's expected.  Shouldn't you checksum it?

Also, is there some problem in just using "busybox-static" instead: https://packages.debian.org/buster/busybox-static ?

Comment (mythi) [reply build/docker/intel-fpga-initcontainer.Dockerfile:59]: > Shouldn't you checksum it?

I'll add the check

Comment (mythi) [reply build/docker/intel-fpga-initcontainer.Dockerfile:59]: @eero-t btw, 893d859138cc0ad38c5fe18736ee372da0de8b99 is from #636

Comment (mythi) [reply build/docker/intel-fpga-initcontainer.Dockerfile]: > This and other initcontainer dockerfiles changes are pulling source tarball from internet without checking it matches what's expected. Shouldn't you checksum it?

Created issue #643 to track this

**Comment by mythi** [comment]
@bart0sh @rojkov ready for review/merge

**Review by uniemimu — COMMENTED** [review]
LGTM

**Review by rojkov — APPROVED** [review]

**Comment by rojkov** [comment]
@kad could you please merge?

**Review by kad — APPROVED** [review]
lgtm

**Pull request merged**